### PR TITLE
Analytics: Add Chart to Revenue Report

### PR DESCRIPTION
**Pull request by timmyc**
For #97 

This branch seeks to be the first step in bringing visualizations to the Revenue Report page. My intentions with this PR are two-fold:

1. The Revenue Report needs a chart to ship, so here is a chart!
2. So far we haven't used the chart component in a "real" scenario yet, so I felt doing the least amount of work to get a chart working on a report page would help surface additional issues needed to complete #164 

With that in mind, please disregard some of the interesting visual glitches in this first pass at adding in the visualization. Pretty please 😄 

image

mobile-revenue

__What is Working__
In this branch you can now see a chart! Super! The chart is using some mock test data, so no live data via the API using the Date Picker params yet. You can also interact with the Summary Numbers to change the chart data series. This was built upon using the same query param logic that already existed in the component - so charts can be _deep_ linked from other pages within wc-admin.

__What Needs Some <3__
As you can see in the screenshot above, there are some definite visual glitches. I'm kind of thinking the chart component may need to implement its own `<Card />` wrapper of sorts, and this would also allow fully building out the chart header with the ability to toggle between line and bar charts, and have the Ellipsis Menu to toggle colors and such ( /cc @greenafrican ).

Though we also need to consider "lite" charts that will be shown in the dashboard which won't need as sophisticated of a header.

I also left some notes in the code for some areas that I think could be good candidates to extract into re-usable components in other charts. Perhaps we can even get to a place where we have a base report component that we simply pass in a config object, and reports magically create themselves. That would be fantastic. Anyhow I'm keen to hear some thoughts on the notes, and just feedback on the implementation.

__To Test__
- load up the branch and navigate to Analytics | Revenue

### CODE REVIEW

File: client/analytics/report/revenue/index.js [hunk 1]
```diff
@@ -229,28 +297,10 @@ class RevenueReport extends Component {
...
-						label={ __( 'Coupons', 'wc-admin' ) }
-						delta={ 15 }
-					/>
-					<SummaryNumber
-						value={ formatCurrency( totals.taxes ) }
-						label={ __( 'Taxes', 'wc-admin' ) }
-					/>
-				</SummaryList>
+				{ this.getChartSummaryNumbers() }
+				<Card title="">
```
Comment (justinshreve) [inline client/analytics/report/revenue/index.js]: I think we should just drop the card wrapper here for now. The `Chart` already contains the header, and this just adds an extra blank box.

**Review by emdashcodes — APPROVED** [review]
> I'm kind of thinking the chart component may need to implement its own <Card /> wrapper of sorts

I think we can some of this functionality to `Chart` itself. It already has the left side of the chart header (with the title and basic legend). We can just build in the chart switcher, ellipsis menu, and the "by $period" switcher (and provide some props for toggling this as well).

> Perhaps we can even get to a place where we have a base report component that we simply pass in a config object, and reports magically create themselves. 

💯. All the elements are related (summary numbers + date picker, etc) and as long as the endpoints return similar data ( `totals`, `intervals`), we should be able to build it out so you pass in some props and we handle the rest. At least for reports that follow a pattern.

So far, I see a wrapper component for summary numbers + chart based on what you did here, and another wrapper for a general report as well.

I'm sure we'll learn a lot from implementing a second one to see what can be abstracted and go from there.

--

I'm pre-approving this, since I think any additional refactoring / creation of components based on this can be done in follow-up PRs. My other comments are minor. This is testing 👍 in browser for me as well.

**Comment by timmyc** [comment]
Thanks for the review @justinshreve - I'll merge this in and look at helping with some of the updates to the base chart component, and creating an issue for the wrapper around summary numbers + chart too.

File: client/analytics/report/revenue/index.js [hunk 2]
```diff
@@ -212,12 +233,59 @@ class RevenueReport extends Component {
...
+			};
+
+			return (
+				<SummaryNumber
+					key={ key }
+					value={ value }
+					label={ label }
+					selected={ isSelected }
+					delta={ 0 }
+					onToggle={ onClick }
```
Comment (ryelle) [inline client/analytics/report/revenue/index.js]: `onToggle` wasn't meant to be used like this, the idea was that each SummaryNumber would use the `href` prop with `getNewPath` so it's a link, not a button.

Comment (timmyc) [reply client/analytics/report/revenue/index.js]: doh! i'll update in a follow-up PR.

**Pull request merged**